jetty-server/src/main/java/org/eclipse/jetty/server/handler/ErrorHandler.java

```diff
@@ -429,23 +434,31 @@ private void writeErrorPlain(HttpServletRequest request, PrintWriter writer, int
 
     private void writeErrorJson(HttpServletRequest request, PrintWriter writer, int code, String message)
     {
-        writer
-            .append("{\n")
-            .append("  url: \"").append(request.getRequestURI()).append("\",\n")
-            .append("  status: \"").append(Integer.toString(code)).append("\",\n")
-            .append("  message: ").append(QuotedStringTokenizer.quote(message)).append(",\n");
+        Throwable cause = (Throwable)request.getAttribute(Dispatcher.ERROR_EXCEPTION);
         Object servlet = request.getAttribute(Dispatcher.ERROR_SERVLET_NAME);
+        Map<String,String> json = new HashMap<>();
+
+        json.put("url", request.getRequestURI());
+        json.put("status", Integer.toString(code));
+        json.put("message", message);
         if (servlet != null)
-            writer.append("servlet: \"").append(servlet.toString()).append("\",\n");
-        Throwable cause = (Throwable)request.getAttribute(Dispatcher.ERROR_EXCEPTION);
+        {
+            json.put("servlet", servlet.toString());
+        }
         int c = 0;
         while (cause != null)
         {
-            writer.append("  cause").append(Integer.toString(c++)).append(": ")
-                .append(QuotedStringTokenizer.quote(cause.toString())).append(",\n");
+            json.put("cause" + c++, cause.toString());
             cause = cause.getCause();
         }
-        writer.append("}");
+
+        writer.append(json.entrySet().stream()
+                .map(e -> QuotedStringTokenizer.quote(e.getKey()) +
+                        ":" +
+                        QuotedStringTokenizer.quote((e.getValue())))
```

Comment:
Assumes that all the values are String, which may not be the case.

Reply:
I guess this json is very simple key and a value. Not sure if we want complex tree here.

Reply:
I agree with @olamy If we keep this `<String,String>` we can generate ourselves.  Anything more complex and we should add a JSON generator dependency.

Reply:
Oook.